jbmc/src/java_bytecode/java_types.h

```diff
@@ -245,6 +245,36 @@ inline bool can_cast_type<java_class_typet>(const typet &type)
 class java_method_typet : public code_typet
 {
 public:
+  using code_typet::parameterst;
+  using code_typet::parametert;
+
+  /// Constructs a new code type, i.e. method type
+  /// \param _parameters: the vector of method parameters
+  /// \param _return_type: the return type
+  java_method_typet(parameterst &&_parameters, typet &&_return_type)
+  {
+    set(ID_C_java_method_type, true);
+    parameters().swap(_parameters);
+    return_type().swap(_return_type);
+  }
+
+  /// Constructs a new code type, i.e. method type
+  /// \param _parameters: the vector of method parameters
+  /// \param _return_type: the return type
+  java_method_typet(parameterst &&_parameters, const typet &_return_type)
```

Comment:
:bulb: One thing that would make this really swell is if the constructor also took the exceptions it might throw (and then ditch `add_throws_exceptions`). Not sure if feasible, but if you did this you would be able to guarantee that all java methods are in the correct state of having their exceptions list.

Reply:
This does sound swell, but is not immediately obvious to me how to do this, not even sure that it is possible.